source/plugin/data/serialization.rst

```diff
@@ -123,3 +125,56 @@ from persistent files using the :doc:`Configurate Library <../configuration/inde
         final DataView container = translator.translate(node);
         return deserializeHealth(container);
     }
+
+
+DataFormat
+==========
+An alternative to DataTranslators is to use :javadoc:DataFormat, which allows you to store a ``DataContainer`` as a HOCON, JSON or NBT file. You can also retrieve data using ``DataFormat``.
+
+This is very useful if you're for example using a database to store information, as you can then serialize any ``DataContainer`` to json format.
+
+For this example we will use :javadoc:`DataFormats#JSON` to translate a ItemStack to json and back! 
+
+**Imports for code examples**
+    import org.spongepowered.api.data.DataContainer;
+    import org.spongepowered.api.data.persistence.DataFormat;
+    import org.spongepowered.api.data.persistence.DataFormats;
+    import org.spongepowered.api.entity.living.player.Player;
+
+    import java.io.ByteArrayOutputStream;
+    import java.io.IOException;
+    import java.io.OutputStream;
+    import java.io.InputStream;
+    import java.util.Optional;
+
+**Code Example: Serializing a ItemStack to json format**
+    
+.. code-block:: java
+
+    public ByteArrayOutputStream toJsonFormat(ItemStack itemStack){
+        DataContainer playerDataContainer = itemStack.toContainer();
+        DataFormat dataFormat = DataFormats.JSON;
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            dataFormat.writeTo(outputStream, playerDataContainer);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return outputStream;
+    }
+
+**Code Example: Serializing a ItemStack from json format**
```

Comment:
That's not what this example is - it's deserializing into a `DataContainer` from `JSON`.

I think what Deamon was getting at is this full example doesn't really tell you much, all you are doing is wrapping the result in an optional, returning an empty `Optional` if it fails. It doesn't tell you much - you might be better served showing how you then could use the resultant `DataContainer`, perhaps showing how to re-create the `ItemStack` from the container? That would fit nicely with your example above - it's showing that the returned `DataContainer` could be a valid `ItemStack` - which it would be in the context of the previous example.